src/string.cr

```diff
@@ -563,13 +563,18 @@ class String
     !!index(str)
   end
 
-  def split
+  def split(limit = nil : Int32?)
+    if limit && limit <= 1
```

Comment:
Yes I agree, but I just kept the style that was already used for the variant that already supported the limit.